### PR TITLE
security/acme-client: Add MailinaBox to Let's Encrypt Service Plugin

### PR DESCRIPTION
**Pull request by billgertz**
Hello OPNsense developers,

Here is my fist wack at submitting a change to your project. Understand I'm new to this project and may have broken with your submission conventions. If I have please let me know, and I'll quickly set things right.

These changes to AcmeClient.xml are primarily to add support for DNS-01 authentication using MailinaBox's Custom DNS http interface. Also added another change to support debug logging from DNSAPI scripts.

There are four file changes to Let's Encrypt plugin:
- certhelper.php in scripts
- AcmeClient.xml in model
- dialogValidation.xml in controller forms
- settings.xml in controllers forms

Please let me know if I'm on the right track or have missed something important. Can report the changes work on my current test system and I'm able to issue certificates using MailinaBox as the target for DNS-01 validation.

Adding dns_miab.sh DNS API as a pull under that repository.

### CODE REVIEW

File: security/acme-client/src/opnsense/mvc/app/models/OPNsense/AcmeClient/AcmeClient.xml [hunk 1]
```diff
@@ -105,6 +105,7 @@
                     <normal>normal</normal>
                     <extended>extended</extended>
                     <debug>debug</debug>
+                    <dnsapi>debug dnsapi</dnsapi>
```
Comment (fraenki) [inline security/acme-client/src/opnsense/mvc/app/models/OPNsense/AcmeClient/AcmeClient.xml]: This change should be a new PR to separate concerns :)

File: security/acme-client/src/opnsense/scripts/OPNsense/AcmeClient/certhelper.php [hunk 2]
```diff
@@ -327,9 +327,17 @@ function eval_optional_acme_args()
...
+    switch($configObj->OPNsense->AcmeClient->settings->logLevel) {
+        case "extended":
+            $acme_args[] = "--log-level 2";
+        case "debug":
+            $acme_args[] = "--debug";
+        case "dnsapi":
+            $acme_args[] = "--debug 2";
+        default:
+            $acme_args[] = "--log-level 1";
+     }
```
Comment (fraenki) [inline security/acme-client/src/opnsense/scripts/OPNsense/AcmeClient/certhelper.php]: Same here, please move this change to a new PR to separate concerns :)

Comment (billgertz) [reply security/acme-client/src/opnsense/scripts/OPNsense/AcmeClient/certhelper.php]: Agreed. As stated in a comment to Frank Wall there is probably a better way of doing this. That is have a separate selection come if debug is selected and then set the debug level from there. Seems that the debug levels are somewhat open ended. Looks like a level 3 is also possible. Playing with debugging I saw that the dnsapi scripts __debug_ lines only show up at _--debug 2_. However you can declare a __debug2_ in dnsapi scripts. Think those debug lines won't be logged unless you select _--debug 3_. I'll give it a test and post back.

**Comment by billgertz** [comment]
Frank.

Saying this here as it may be buried in the code comments. 

There is probably a better way to do the debug level change. That is have a separate selection come if debug is selected and then set the debug level from there. Seems that the debug levels are somewhat open ended. Looks like a level 3 is also possible. Playing with debugging I saw that the dnsapi scripts _debug lines only show up at --debug 2. However you can declare a _debug2 in dnsapi scripts. Think those debug lines won't be logged unless you select --debug 3. I'll give it a test and post back.

File: security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/settings.xml [hunk 3]
```diff
@@ -1,47 +0,0 @@
-<form>
```
Comment (fraenki) [inline security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/settings.xml]: oops, I think you should not remove this file :)

Comment (billgertz) [reply security/acme-client/src/opnsense/mvc/app/controllers/OPNsense/AcmeClient/forms/settings.xml]: Tried t remote it from the commit using the web interface - my bad. Do I just read the original back?

**Comment by billgertz** [comment]
Frank,

Looks like I got put back right. The number of files changed is now three and settings.xml doesn't show up in the diff.

Sorry about that. Should be using this from the cli and not the web interface.

**Comment by fraenki** [comment]
@billgertz LGTM now. Thanks for your contribution!

I'll wait for the merge until the required changes are available in an acme.sh release.

**Comment by billgertz** [comment]
Frank - thanks got one last bug in the dns_miab.sh I'm working on. To the debug level PR which way should we go as I originally proposed (integrated) or selecting debug brings up a new option to select level. Since acme.sh takes _--debug_ and _--debufg 2_, I would suggest the integrated approach.

**Comment by billgertz** [comment]
@fraenki

Neil Pang just merged the MailinaBox DNS-01 API script to acme.sh. Should show up with next release. See: https://github.com/Neilpang/acme.sh/pull/2534

**Comment by billgertz** [comment]
@fraenki 
Looks like Neil Pang just merged dns_miab.sh to current version. Should be available when updating or new install of acme.sh.

**Comment by fraenki** [comment]
@billgertz Thanks, we're almost there. :) We'll just have to wait for the next official release of amce.sh.

**Comment by fraenki** [comment]
@billgertz There's still no official acme.sh release available, but I've merged it anyway. Thanks for your contribution!

(This means that it's not working out of the box right now, but the missing acme.sh dnsapi file can still be copied manually.)

**Comment by billgertz** [comment]
Frank,

Thanks! You already probably know running:

acme.sh --upgrade

would also install the newer DNSAPI scripts, including dns-miab.sh

Regards,

Bill

**Pull request merged**